CRM/Case/BAO/Case.php

```diff
@@ -1253,7 +1253,7 @@ public static function getRelatedContacts($caseID, $includeDetails = TRUE) {
           'email' => $dao->email,
         );
         // Add more info about the role (creator, manager)
-        $role = CRM_Utils_Array::value($details['role'], $caseRoles);
+        $role = CRM_Utils_Array::value($dao->name_b_a, $caseRoles);
```

Comment:
I was concerned about this change being a semi-arbitrary design-change, but on closer look it does seem like a bug fix.

To support this interpretation, one should ask "What data is in `$caseRoles`?" If you go up a few lines, the content is determined by:

```php
$caseRoles = CRM_Utils_Array::rekey($caseInfo['case_type_id.definition']['caseRoles'], 'name');
```

which means that `$caseRoles` is specifically keyed by `name` instead of `label`. This signifies that prior revisions were inconsistent about whether to use `name` or `label`, which would be manifestly buggy. This makes it consistent.